### DISCUSSION TITLE
p:schedule allday events over multiple days - last day not shown as expected

### DISCUSSION THREAD

**Discussion by mh1985** [primefaces/community#648] [PrimeFaces] [answered]
When an event marked as 'allday' has different dates for the start and end, the last day is not displayed on the calendar. The event appears one day shorter.

It's not an issue to compensate for this in the backing bean, but the question is: is this intentional or is it a bug?

![schedule_allday_bug_1](https://github.com/primefaces/community/assets/9570716/6f19e4a8-0d24-41c2-85c6-2dc94ad793cf)
![schedule_allday_bug_2](https://github.com/primefaces/community/assets/9570716/c0972178-67ba-4bd4-adc7-2b23c3017ba7)
![schedule_allday_bug_3](https://github.com/primefaces/community/assets/9570716/5c0acd77-68b2-4181-965f-21850f474ace)

I know, that it's logical when the event is until 21/12/2023 00:00, but it's not really intuitive to use, as a user I would expect to include the "To" date. That is the reason the time is hidden in my project, if the user wants to set an event as allday.

![schedule_allday_bug_4](https://github.com/primefaces/community/assets/9570716/f3155b7e-9fef-42ea-a557-926627f39dc8)
![schedule_allday_bug_5](https://github.com/primefaces/community/assets/9570716/984fed1b-2245-400d-bf8a-40060d8bda54)

**Comment by melloware** [comment, marked as answer]
I think your issue is this which turns out how FullCalendar works: https://github.com/primefaces/primefaces/issues/1164

**Reply by mh1985** [reply]
Thanks. Didn't think about looking into FullCalendar for this. I'll correct the date in my backing bean then.

**Reply by melloware** [reply]
At least you are not alone in thinking it should have worked the way you expected but there are multiple threads on fullcalendar where they explain WHY they do it that way so who knows!
